File: languages/php/Bloock/Entity/Key/AccessControlSecret.php
<?php

namespace Bloock\Entity\Key;

/**
 * Represents a secret used for secret-based access control.
 */
class AccessControlSecret
{
    public string $secret;

    /**
     * Creates a new AccessControlSecret instance with the provided secret.
     * @param string $secret
     */
    public function __construct(string $secret)
    {
        $this->secret = $secret;
    }

    public function toProto(): \Bloock\AccessControlSecret
    {
        $s = new \Bloock\AccessControlSecret();

        $s->setSecret($this->secret);

        return $s;
    }
}

File: languages/python/bloock/entity/key/access_control_secret.py
from __future__ import annotations

import bloock._bridge.proto.keys_entities_pb2 as proto


class AccessControlSecret:
    """
    Represents a secret used for secret-based access control.
    """
    secret = None

    def __init__(self, secret) -> None:
        """
        Creates a new AccessControlSecret instance with the provided secret.
        :type secret: object
        :rtype: object
        """
        self.secret = secret

    def to_proto(self) -> proto.AccessControlSecret:
        return proto.AccessControlSecret(
            secret=self.secret
        )